superset/models/core.py

```diff
@@ -761,8 +761,11 @@ def wrapper(*args, **kwargs):
             post_data = request.form or {}
             d.update(post_data)
             d.update(kwargs)
-            slice_id = d.get('slice_id', 0)
             try:
+                slice_id = d.get('slice_id') 
+                form_data = d.get('form_data')
+                if not slice_id and form_data:
+                    slice_id = json.loads(form_data).get('slice_id')                    
                 slice_id = int(slice_id) if slice_id else 0
             except ValueError:
```

Comment:
json.loads might raise something else than `ValueError`, for instance `json.loads(None)` raises `TypeError`. You might just want to be safe and catch all exceptions here as we're not trying to handle a specific type of error but just keep the program flowing regardless of what error may be raised.